marimo/_server/session/session_view.py

```diff
@@ -156,6 +166,19 @@ def get_cell_console_outputs(
                 outputs[cell_id] = as_list(cell_op.console)
         return outputs
 
+    def set_execution_time(
+        self, cell_id: CellId_t, event: Literal["start", "end"]
+    ) -> None:
+        """Updates execution time for given cell."""
+        if event == "start":
+            time_elapsed = time.time()
```

Comment:
We have a timestamp on the operation. Probably the same - but better to use that one. It will work better in tests since we could mock out those times instead of these ones.